src/components/container.js

```diff
@@ -151,22 +157,15 @@ export default class Container extends Component {
     };
   }
 
-  triggerScopeChange(prevScopeId) {
-    const { storeType } = this.constructor;
-    const previous = this.getScopedStore(storeType, prevScopeId);
-    // When called, subscribers that have already re-rendered with the new
-    // scope are no longer subscribed to the old one, so we "force update"
-    // the remaining.
-    // This is sub-optimal as if there are other containers with the same
-    // old scope id we will re-render those too, but better than using context
-    // as that will re-render all children even if pure/memo
-    previous.storeState.notify();
-  }
-
-  deleteScopedStore(scopeId = this.props.scope) {
-    const { storeType } = this.constructor;
-    const { storeState } = this.getScopedStore(storeType, scopeId);
-    if (scopeId != null && !storeState.listeners().length) {
+  deleteScopedStore(prevStoreState, scopeId = this.props.scope) {
+    const { storeState } = this.getScopedStore(scopeId);
+    if (
+      scopeId != null &&
+      !prevStoreState.listeners().length &&
+      // ensure registry has not already created a new store w/ same scope
+      prevStoreState === storeState
+    ) {
+      const { storeType } = this.constructor;
       this.getRegistry().deleteStore(storeType, scopeId);
```

Comment:
I have a strong feeling that the guard rail should be in `deleteStore`, not React component.

Reply:
Maybe. Currently the registry does not keep track of the stores lifecycle, so if we ask to delete something why should it say no? 
We could shift that management to the Registry itself, but would need to implement some signals/comms layer to let registry know that store subscribers have changed, that a container has been unmounted, ... 
It could be a more solid solution, but I feel it might be too elaborate. At the end of the day this is an edge case, and maybe we can handle it better when we will rewrite this class into a functional component.